### PR TITLE
Make %last_error% object

### PR DESCRIPTION
**Pull request by mikevoronov**

### CODE REVIEW

File: air/src/execution/utils/resolve.rs [hunk 1]
```diff
@@ -53,27 +55,18 @@ fn prepare_consts(arg: impl Into<JValue>, ctx: &ExecutionCtx<'_>) -> ExecutionRe
...
-        }
+fn prepare_last_error(
+    path: &LastErrorPath,
+    ctx: &ExecutionCtx<'_>,
+) -> ExecutionResult<(JValue, Vec<SecurityTetraplet>)> {
+    let LastErrorWithTetraplets { last_error, tetraplets } = ctx.last_error();
+    let jvalue = match path {
+        LastErrorPath::Instruction => JValue::String(last_error.instruction),
+        LastErrorPath::Message => JValue::String(last_error.msg),
+        LastErrorPath::None => serde_json::to_value(last_error).expect("the default serializer shouldn't fail"),
```
Comment (folex) [inline air/src/execution/utils/resolve.rs:66]: you can do `json!(last_error)`. It panics inside, but looks nicer :D

File: crates/air-parser/src/parser/lexer/air_lexer.rs [hunk 2]
```diff
@@ -194,6 +195,25 @@ fn string_to_token(input: &str, start_pos: usize) -> LexerResult<Token> {
     }
 }
 
+fn parse_last_error(input: &str, start_pos: usize) -> LexerResult<Token<'_>> {
+    let last_error_size = LAST_ERROR.len();
+    let last_error_path = match &input[last_error_size..] {
```
Comment (folex) [inline crates/air-parser/src/parser/lexer/air_lexer.rs:200]: Maybe also account for possible `$`? Like `%last_error%.$.instruction`

**Review by folex — APPROVED** [review]

File: air/src/execution/utils/resolve.rs (hunk 1 above)
Comment (mikevoronov) [reply air/src/execution/utils/resolve.rs:66]: Yep, will do, it really looks nicer.

File: crates/air-parser/src/parser/lexer/air_lexer.rs (hunk 2 above)
Comment (mikevoronov) [reply crates/air-parser/src/parser/lexer/air_lexer.rs:200]: Hmm, I am not sure, we're going to get rid of json path in the nearest future.

Comment (mikevoronov) [reply crates/air-parser/src/parser/lexer/air_lexer.rs:200]: Discuss it in PM and decided to use `.$` here for consistency.

**Pull request merged**